ConfigurationSystem/Agent/UsersAndGroups.py

```diff
@@ -338,7 +338,7 @@ def __syncCSWithVOMS( self ):
       for k in ( 'DN', 'CA', 'Email' ):
         csUserData[ k ] = ", ".join( csUserData[ k ] )
       result = csapi.modifyUser( user, csUserData, createIfNonExistant = True )
-      if not result[ 'OK' ]:
+      if not result[ 'OK' ] or not result['Value']:
         self.__adminMsgs[ 'Error' ].append( "Cannot modify user %s: %s" % ( user, result[ 'Message' ] ) )
```

Comment:
If you enter here when result = {'OK': True, 'Value': ''} then accessing result['Message'] will issue a KeyError